pkg/asset/manifests/tectonic.go

```diff
@@ -93,8 +93,6 @@ func (t *Tectonic) Generate(dependencies asset.Parents) error {
 				Base64encodeCloudCreds: credsEncoded,
 			},
 		}
-	default:
-		return errors.Errorf("unknown provider, could not populate cloud credentials secret data")
 	}
```

Comment:
Don't throw error here, just generate an empty secret.  As we add different cloud-providers, they may/may not require credentials.  If they don't/we don't add any,  installer won't break if we don't default to error here.  It's not the responsibility to check for accepted cloud-provider here, anyways.